agent/structs/system_metadata.go

```diff
@@ -0,0 +1,36 @@
+package structs
+
+// SystemMetadataOp is the operation for a request related to system metadata.
+type SystemMetadataOp string
+
+const (
+	SystemMetadataOpUpsert SystemMetadataOp = "upsert"
+	SystemMetadataOpDelete SystemMetadataOp = "delete"
+)
+
+// SystemMetadataRequest is used to upsert and delete system metadata.
+type SystemMetadataRequest struct {
+	// Datacenter is the target for this request.
+	Datacenter string
+
+	// Op is the type of operation being requested.
+	Op SystemMetadataOp
+
+	// Entries is the set of keys to modify.
+	Entries []*SystemMetadataEntry
```

Comment:
How come this accepts a slice rather than a single entry?

Reply:
I did not have a use case for multiple atomic updates at this time. It seemed somewhat cheap to make it possible to do multiple atomic updates in the future without having to resort to introducing either another raft operation integer code or to shim this into `Txn`.

Alternatively I could do it this way now:
```
type SystemMetadataRequest struct {
  Datacenter String
  Op SystemMetadataOp
  Entry *SystemMetadataEntry
  WriteRequest
}
```
And then should we need multi-write, we could just do it union style:
```
type SystemMetadataRequest struct {
  Datacenter String
  Op SystemMetadataOp

  // Only one of Entry or Entries may be specified.
  Entry *SystemMetadataEntry
  Entries []*SystemMetadataEntry

  WriteRequest
}
```
Thoughts?

Reply:
Yeah that is a possibility. I lean toward having a single entry per request and revisiting the txn path later.